### ISSUE TITLE
The podgroup is not deleted after the pod created by Deployment is deleted 

### ISSUE THREAD

**Issue by hansongChina** [volcano-sh/volcano#1853]
**What happened**:
When changing Deployment triggers the creation of a new ReplicaSet and the pod created by the old ReplicaSet is deleted, the podgroup of the old pod is not deleted

**What you expected to happen**:

the podgroup of the old pod is deleted

**How to reproduce it (as minimally and precisely as possible)**:

- step1:Add the default volcano to pod using admission Webhooks, for example

`patches = append(patches, patchOperation{
			Op:    "add",
			Path:  "/spec/schedulerName",
			Value: "volcano",
		})`
- step2:Create a Deployment,for example
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  labels:
    app: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.14.2
        ports:
        - containerPort: 80
```
```shell
kubectl apply -f nginx-deployment.yaml
```
![image](https://user-images.githubusercontent.com/7590000/143237845-f6fecac1-f7fc-4673-9b72-616f5785cba1.png)
```shell
kubectl edit pod nginx-deployment-6b474476c4-v6k4x
```
![image](https://user-images.githubusercontent.com/7590000/143237933-6709bed5-d3b1-4ae2-b90d-534b2f0ce9ce.png)
```shell
kubectl edit pg  podgroup-6f08b2ae-296e-43a1-825c-84e483c3ea89
```
![image](https://user-images.githubusercontent.com/7590000/143238061-72dc2155-8298-4a43-bf6e-9aa2bc20a917.png)
- step3: Modify "image:nginx:1.14.2" in Deployment to trigger the creation of a new RS and a new POD. The old pod is deleted and the podgroup associated with the old POD still exists in the state of inqueue
![image](https://user-images.githubusercontent.com/7590000/143239373-fcd60614-ba7a-43d5-9f27-0d7875b98376.png)
```shell
kubectl edit pg  podgroup-6f08b2ae-296e-43a1-825c-84e483c3ea89
```
![image](https://user-images.githubusercontent.com/7590000/143240533-ba236ce3-d069-4b2f-a91f-2e35ea65bc36.png)

**Anything else we need to know?**:

<font color=red>kind is workflow runs into the same problem</font>

**Environment**:
- Volcano Version: v1.4.0
- Kubernetes version (use `kubectl version`): 
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.9", GitCommit:"94f372e501c973a7fa9eb40ec9ebd2fe7ca69848", GitTreeState:"clean", BuildDate:"2020-09-16T13:56:40Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.9", GitCommit:"94f372e501c973a7fa9eb40ec9ebd2fe7ca69848", GitTreeState:"clean", BuildDate:"2020-09-16T13:47:43Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration:
![image](https://user-images.githubusercontent.com/7590000/143236161-865459d3-de94-4de3-adc5-b54a52c54308.png)
![image](https://user-images.githubusercontent.com/7590000/143236292-1883eacd-c485-477a-8ffe-4e1c0f6d7a6c.png)

- OS (e.g. from /etc/os-release):
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
- Kernel (e.g. `uname -a`):
Linux  3.10.0-957.12.2.el7.x86_64 #1 SMP Tue May 14 21:24:32 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Others:

**Comment by hansongChina** [comment]
**kind is workflow runs into the same problem**

**Comment by k82cn** [comment]
For `Deployment`, we should create PodGroup for `Deployment` instead of `ReplicaSet` :)

**Comment by D0m021ng** [comment]
Normal Pod runs into the same problem.

**Comment by Thor-wl** [comment]
Thanks for the all the feedback. This fix will be merged into v1.5.0.
/assign @lucming

**Comment by Thor-wl** [comment]
> Normal Pod runs into the same problem.

Can you give the reproduce steps? I've had a try but the performance looks good.
